contrib/portable_realpath_fun.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# re-implementation of realpath, because Mac OS doesn't have realpath
+# https://stackoverflow.com/questions/3572030/bash-script-absolute-path-with-os-x/18443300#18443300
+portable_realpath() {
+    local ourpwd=$PWD
+    cd "$(dirname "$1")"
+    local link=$(readlink "$(basename "$1")")
+    while [ "$link" ]; do
+        cd "$(dirname "$link")"
+        link=$(readlink "$(basename "$1")")
+    done
+    local ourrealpath="$PWD/$(basename "$1")"
+    cd "$ourpwd"
+    echo "$ourrealpath"
+}
```

Comment:
Not sure if this script is actually doing what it should. For example, for path `/foo/bar/cvc4/contrib/../..` above script produces `/foo/bar/cvc4/..`, whereas realpath produces `/foo/bar`. So why not just use `$(pwd)/../..` as prefix and don't use realpath at all? If we want to resolve symbolic links, I would actually vote for the Python approach. Sure we have to call the Python interpreter, but it is less error prone than bash hacks.

Reply:
Thanks for checking that -- should have unit-tested it. Based on comments it looks like we should just use an absolute path to avoid warnings so I went the python route.